polygeist/tools/cgeist/Lib/clang-mlir.h

```diff
@@ -521,7 +521,13 @@ class MLIRScanner : public clang::StmtVisitor<MLIRScanner, ValueCategory> {
   ValueCategory VisitBin##OP(clang::BinaryOperator *E);                        \
   ValueCategory VisitBin##OP##Assign(clang::BinaryOperator *E);
 #include "Expressions.def"
-#undef HANDLEBINOP
+
+#define HANDLEUNARYOP(OP)                                                      \
+  ValueCategory VisitUnary##OP(clang::UnaryOperator *E,                        \
+                               clang::QualType PromotionTy =                   \
+                                   clang::QualType());                         \
+  ValueCategory Visit##OP(clang::UnaryOperator *E, clang::QualType PromotionTy);
+#include "Expressions.def"
```

Comment:
After macro has been used we should #undef it.

Reply:
It's being `#undef`ed in the `Expressions.def` file.

Reply:
Ok all good then.